### ISSUE TITLE
Better handling of anything between "next year" to "someday

### ISSUE THREAD

**Issue by evilham** [AnotherKamila/tasksched#24]
Follow up for #22 (thank you @robertbasden !).

Tasksched shows only "Month date", and "someday" for anything over 10 years; which can be confusing if something is scheduled on say, March 20th 2019 (today being February 2018).

Maybe we want to have 4 tiers of date formatting:

* Month date: for anything under 1 year
* "Next year" for anything between 1-2 years
* "In X years" for X years, with X between 2-10 years
* "Someday" for anything over 10 years

@AnotherKamila mentioned that these conditionals should go out of `TaskView.elm` as a date formatting function, she'll probably add details on that :).

**Comment by AnotherKamila** [comment]
Indeed, a refactoring would be good: instead of having `if`s in the HTML, the `TaskView.elm` functions should use functions from `Date/Human.elm` (which already accept a `now` parameter). The stuff in `Date/Human.elm` should be expanded to handle this use.
